src/ProjectService.Data/ProjectRepository.cs

```diff
@@ -131,13 +114,21 @@ public List<DbProject> FindProjects(FindDbProjectsFilter filter, int skipCount,
             }
 
             var dbProjects = _provider.Projects
-                .AsSingleQuery()
                 .AsQueryable();
 
-            var projects = CreateFindPredicates(filter, dbProjects).ToList();
-            totalCount = projects.Count;
+            if (filter.DepartmentId.HasValue)
+            {
+                dbProjects = dbProjects.Where(p => p.DepartmentId == filter.DepartmentId.Value);
+                totalCount = dbProjects.Count(p => p.DepartmentId == filter.DepartmentId.Value);
+            }
+            else
+            {
+                totalCount = dbProjects.Count();
+            }
+
+            dbProjects = dbProjects.Skip(skipCount).Take(takeCount);
 
-            return projects.Skip(skipCount).Take(takeCount).ToList();
+            return dbProjects.ToList();
```

Comment:
may be better write all inline  `return dbProjects.Skip(skipCount).Take(takeCount)`

Reply:
fixed